bindgen/ir/var.rs

```diff
@@ -389,9 +389,60 @@ impl ClangSubItemParser for Var {
     }
 }
 
+/// This function uses a [`FallbackTranslationUnit`][clang::FallbackTranslationUnit] to parse each
+/// macro that cannot be parsed by the normal bindgen process for `#define`s.
+///
+/// To construct the [`FallbackTranslationUnit`][clang::FallbackTranslationUnit], first precompiled
+/// headers are generated for all input headers. An empty temporary `.c` file is generated to pass
+/// to the translation unit. On the evaluation of each macro, a [`String`] is generated with the
+/// new contents of the empty file and passed in for reparsing. The precompiled headers and
+/// preservation of the [`FallbackTranslationUnit`][clang::FallbackTranslationUnit] across macro
+/// evaluations are both optimizations that have significantly improved the performance.
+fn parse_macro_clang_fallback(
+    ctx: &mut BindgenContext,
+    cursor: &clang::Cursor,
+) -> Option<(Vec<u8>, cexpr::expr::EvalResult)> {
+    if !ctx.options().clang_macro_fallback {
+        return None;
+    }
+
+    let file = format!(
```

Comment:
The translation unit has the file path already, so this could be moved to `try_ensure_fallback_translation_unit`, and just use `ftu.file_path` below, right?

Reply:
This depends a little bit on how you want to see this implemented. On one hand, we could make the member of `FallbackTranslationUnit` crate public, but I think we'd require a clone because otherwise when we pass it in, we'd have a mutable reference taken for `.reparse()` and an immutable reference taken for `&ftu.file_path`. We could also change the API slightly so that `.reparse()` only takes file contents as it's only operating on a single file and we may not need the flexibility of supporting reparsing multiple different files. I'd personally prefer the second approach.

Reply:
I've made the changes to reflect the second option. Let me know what you think.